blog/2022-11-09-nginx-monitoring.md

```diff
@@ -0,0 +1,102 @@
+- **requests per second** - which provides a high-level view of combined end-user activity
+- **server error rate** - which indicates how often your servers are failing to process valid requests and 
+- **request processing time** - which describes how long your servers are taking to process client requests (and which can point to slowdowns or other problems in your environment).
+
+## NGINX Activity metrics
+
+Irrespective of the NGINX use case, you will always need to monitor how many client requests your servers are receiving and how those requests are being processed.
+The connection requests can be monitored in multiple sub-categories:
+
+- **Requests per second** - Which provides a high-level view of combined end-user activity and the required bandwidth.
+![Requests per second](https://user-images.githubusercontent.com/96257330/200830717-e2df3c9d-c02b-4de5-9a29-685d5ac4a547.png)
+
+- **Active Connections** - Which shows the currently active client connections and is an indication of the resource utilisation.
+!["Active Connections"](https://user-images.githubusercontent.com/96257330/200822049-6aebe009-78bc-48c8-aa2a-83cd90174004.png)
+
+- **Connection Status** - Which provides insights on what these connections are doing (**idle, reading, writing**)
+!["Connection Status](https://user-images.githubusercontent.com/96257330/200822841-60d29358-3e08-4221-a283-6b6a07a1829f.png)
+
+You can additionally group these metrics by instance to see how many connections are being handled per instance and in which state they are.
+![Connection Status per Instance](https://user-images.githubusercontent.com/96257330/200823083-2a35da9f-f6fc-4199-8c64-78b9417d4702.png)
+
+
+- **Accepted vs Handled Connections** - Which indicates if any connections are being dropped by calculating (**Accepted - Handled**) 
+![Accepted vs Handled](https://user-images.githubusercontent.com/96257330/200823600-b4a0c2d3-c430-4f94-b6e8-8ac1b4f2252b.png)
+
+You can also setup a custom alert to report any dropped connections on your NGINX.
+For example, the alert shown below will raise a Warning alert if there are 10-20% dropped connections and a Critical alert when the dropped connections go beyond 20%.
+
+```yaml
+ template: NGINX_Dropped_Connections_Exceeded
+    on: nginx.connections_accepted_handled
+ class: Utilization
+ type: NGINX
+ component: NGINX
+ calc: (($accepted - $handled) * 100) / ($accepted)
+ every: 1m
+ units: %
+  warn: $this > (($status >= $WARNING)  ? (10) : (20))
+  crit: $this > (($status == $CRITICAL) ? (20) : (30))
+  delay: down 15m multiplier 1.5 max 1h
+  info: The NGINX web server has exceeded the limit of dropped connections
+
+```
+## NGINX Error Metrics
+
+NGINX error metrics tell you how often your servers are returning errors instead of producing useful work. Client errors are represented by 4xx status codes, server errors with 5xx status codes.
```

Comment:
```suggestion
NGINX error metrics tell you how often your servers are returning errors instead of producing useful work. Client errors are represented by [4xx status codes](https://en.wikipedia.org/wiki/List_of_HTTP_status_codes#4xx_client_errors), server errors with [5xx status codes](https://en.wikipedia.org/wiki/List_of_HTTP_status_codes#5xx_server_errors).
```